pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -380,61 +392,74 @@ func (m *manager) removeStaleState() {
+				continue
+			}
+
+			lh.V(2).Info("removing container")
+			err := m.policyRemoveContainerByRef(ctx, podUID, containerName)
+			if err != nil {
+				lh.Error(err, "failed to remove container")
 			}
 		}
 	}
 
 	m.containerMap.Visit(func(podUID, containerName, containerID string) {
-		if _, ok := activeContainers[podUID][containerName]; !ok {
-			klog.ErrorS(nil, "RemoveStaleState: removing container", "podUID", podUID, "containerName", containerName)
-			err := m.policyRemoveContainerByRef(podUID, containerName)
-			if err != nil {
-				klog.ErrorS(err, "RemoveStaleState: failed to remove container", "podUID", podUID, "containerName", containerName)
-			}
+		lh := klog.LoggerWithValues(logHandle, "podUID", podUID, "containerName", containerName)
+		if _, ok := activeContainers[podUID][containerName]; ok {
+			lh.V(5).Info("containerMap: container still active")
+			return
+		}
+		lh.V(2).Info("containerMap: removing container")
+		err := m.policyRemoveContainerByRef(ctx, podUID, containerName)
+		if err != nil {
+			lh.Error(err, "containerMap: failed to remove container")
 		}
 	})
 }
 
-func (m *manager) reconcileState() (success []reconciledContainer, failure []reconciledContainer) {
-	ctx := context.Background()
+func (m *manager) reconcileState(ctx context.Context) (success []reconciledContainer, failure []reconciledContainer) {
 	success = []reconciledContainer{}
 	failure = []reconciledContainer{}
 
-	m.removeStaleState()
+	logHandle := klog.LoggerWithName(klog.FromContext(ctx), "reconcileState")
+
+	m.removeStaleState(ctx)
 	for _, pod := range m.activePods() {
+		lh := klog.LoggerWithValues(logHandle, "pod", klog.KObj(pod))
+
 		pstatus, ok := m.podStatusProvider.GetPodStatus(pod.UID)
 		if !ok {
-			klog.V(4).InfoS("ReconcileState: skipping pod; status not found", "pod", klog.KObj(pod))
+			lh.V(5).Info("skipping pod; status not found")
 			failure = append(failure, reconciledContainer{pod.Name, "", ""})
 			continue
 		}
 
 		allContainers := pod.Spec.InitContainers
 		allContainers = append(allContainers, pod.Spec.Containers...)
 		for _, container := range allContainers {
+			lh = klog.LoggerWithValues(lh, "containerName", container.Name)
+
 			containerID, err := findContainerIDByName(&pstatus, container.Name)
 			if err != nil {
-				klog.V(4).InfoS("ReconcileState: skipping container; ID not found in pod status", "pod", klog.KObj(pod), "containerName", container.Name, "err", err)
+				lh.V(5).Info("skipping container; ID not found in pod status", "err", err)
 				failure = append(failure, reconciledContainer{pod.Name, container.Name, ""})
 				continue
 			}
 
 			cstatus, err := findContainerStatusByName(&pstatus, container.Name)
 			if err != nil {
-				klog.V(4).InfoS("ReconcileState: skipping container; container status not found in pod status", "pod", klog.KObj(pod), "containerName", container.Name, "err", err)
+				lh.V(5).Info("skipping container; container status not found in pod status", "err", err)
```

Comment:
same above  Maybe adding more information would be better

Reply:
What should be added?

Reply:
my bad. When I went in and saw the `findContainerStatusByName` method, I found that there was already container name information